Rename JSImmutableButterfly to JSCellButterfly as its no longer always immutable

kmiller68 opened the pull request:
#### b660c35ed3fc50d61998e00ed61f1571bf9d81de
<pre>
Rename JSImmutableButterfly to JSCellButterfly as its no longer always immutable
<a href="https://bugs.webkit.org/show_bug.cgi?id=297677">https://bugs.webkit.org/show_bug.cgi?id=297677</a>
<a href="https://rdar.apple.com/158787973">rdar://158787973</a>

Reviewed by Yusuke Suzuki.

This name is now misleading. Originally it was only used for copy on write butterflies so immutable was accurate.
We&apos;ve since started using JSImmutableButterflies for other non-immutable purposes. This patch changes the name to a
more accurate JSCellButterfly name. The patch is almost entirely a find replace, although I left the variables for
`immutableButterfly` when refactoring the original array literal use case. Other changes are noted below:

* Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp:
    Removed dead check for copy on write for the immutable butterfly. Made it a RELEASE_ASSERT instead.
* Source/JavaScriptCore/runtime/JSCellButterfly.h: Renamed from Source/JavaScriptCore/runtime/JSImmutableButterfly.h.
    Added a comment explaining what a JSCellButterfly is.
* Source/JavaScriptCore/runtime/OrderedHashTableHelper.h:
    Added a FIXME to make the JSCellButterfly not CopyOnWrite since it&apos;s definitely mutated by that data structure.

Canonical link: <a href="https://commits.webkit.org/298974@main">https://commits.webkit.org/298974@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5f70502ad97e30f99e5c9dd1fc6f1f60d2455dc3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117356 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37026 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27643 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123455 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/69343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/34daf635-3eb7-43e2-a56f-747d2563f2f6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37722 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45613 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/89060 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/69343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/31f23a4f-fdcc-4b79-b442-7a23cfcf86b7) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120308 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30023 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105227 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/69567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0b2e8201-7af8-4206-a217-850b166a00e6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29080 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/23341 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67128 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109461 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99424 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23523 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126576 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115863 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44253 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/33254 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44610 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101461 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/97524 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24829 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42880 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/20817 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44126 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49785 "Build is being retried. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Checked out pull request") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144563 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43582 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/37212 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46927 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45278 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5f70502ad97e30f99e5c9dd1fc6f1f60d2455dc3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49669

kmiller68 committed:
Rename JSImmutableButterfly to JSCellButterfly as its no longer always immutable
https://bugs.webkit.org/show_bug.cgi?id=297677
rdar://158787973

Reviewed by Yusuke Suzuki.

This name is now misleading. Originally it was only used for copy on write butterflies so immutable was accurate.
We've since started using JSImmutableButterflies for other non-immutable purposes. This patch changes the name to a
more accurate JSCellButterfly name. The patch is almost entirely a find replace, although I left the variables for
`immutableButterfly` when refactoring the original array literal use case. Other changes are noted below:

* Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp:
    Removed dead check for copy on write for the immutable butterfly. Made it a RELEASE_ASSERT instead.
* Source/JavaScriptCore/runtime/JSCellButterfly.h: Renamed from Source/JavaScriptCore/runtime/JSImmutableButterfly.h.
    Added a comment explaining what a JSCellButterfly is.
* Source/JavaScriptCore/runtime/OrderedHashTableHelper.h:
    Added a FIXME to make the JSCellButterfly not CopyOnWrite since it's definitely mutated by that data structure.

Canonical link: https://commits.webkit.org/298974@main
